controller/src/test/java/io/pravega/controller/task/Stream/StreamMetadataTasksTest.java

```diff
@@ -458,6 +460,76 @@ public void retentionStreamTest() throws Exception {
         assertTrue(truncProp.getProperty().getStreamCut().get(0) == 1L && truncProp.getProperty().getStreamCut().get(1) == 1L);
     }
 
+    @Test(timeout = 30000)
+    public void sizeBasedRetentionStreamTest() throws Exception {
```

Comment:
A similar comment for this test case. It is testing one scenario, and I'm wondering of we can not only test more scenarios, but perhaps introduce some randomization to the test case, like number of segments, position of a cut in the segment, number of scaling events, etc. to increase test coverage.

Reply:
> It is testing one scenario
we test various scenarios in this test case. Covers most logical scenarios where you have  a stream cut set and a truncation point and a policy. 
we test multiple variations of following:
1. no stream cut eligible as new truncation point
2. multiple stream cuts eligible for new truncation point

Reply:
added additional scenarios including interference with manual truncation. So now this test is pretty comprehensive and covers most interesting cases.